src/main/java/com/hubspot/smtp/client/SmtpSession.java

```diff
@@ -53,6 +59,13 @@
       SmtpCommand.QUIT,
       SmtpCommand.NOOP);
 
+  private static final Supplier<Executor> SHARED_DEFAULT_EXECUTOR = Suppliers.memoize(SmtpSession::getSharedExecutor);
+
+  private static ExecutorService getSharedExecutor() {
+    ThreadFactory threadFactory = new ThreadFactoryBuilder().setDaemon(true).setNameFormat("niosmtpclient-%d").build();
+    return Executors.newCachedThreadPool(threadFactory);
```

Comment:
Thanks! I haven't used `allowCoreThreadTimeout` before, but checking the implementation of `newCachedThreadPool` it seems it sets the number of core threads to zero, so `allowCoreThreadTimeout` shouldn't apply I think? I tested this and after the default timeout of 60s there are no threads left in the pool. Maybe I'm looking at the wrong thing though...